src/components/status/list/content.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState, useEffect } from 'react';
+import classnames from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+interface IContent {
+  index: number;
+  nowIndex: number;
+  changeIndex: (display: number) => void;
+  bidType: string;
+  status: string;
+  date: string;
+  time: string;
+  price: number;
+  hash: string;
+}
+
+const Content: React.FC<IContent> = ({
+  index,
+  nowIndex,
+  changeIndex,
+  bidType,
+  status,
+  date,
+  time,
+  price,
+  hash,
+}) => {
+  const { t } = useTranslation();
+
+  // change button color and pass state
+  const button = classnames('status-list-content-button');
+  const buttonClick = classnames(
+    'status-list-content-button',
+    'status-list-content-button--click',
+  );
+  const [buttonColor, setButtonColor] = useState<string>(button);
+  const [color, setColor] = useState<boolean>(false);
+  useEffect(() => {
+    setButtonColor(color === false ? button : buttonClick);
+  }, [color]);
+
+  useEffect(() => {
+    if (nowIndex !== index) setColor(false);
+  }, [nowIndex]);
+
+  const infoOnClick = () => {
+    setColor(true);
+    changeIndex(index);
+  };
+
+  // bidType text
+  const bidTypeSell = classnames(
+    'status-list-content-bidTypeSell',
+    'status-list-content-bidType',
+  );
+  const bidTypeBuy = classnames(
+    'status-list-content-bidTypeBuy',
+    'status-list-content-bidType',
+  );
+  const bidTypeNeed = classnames(
+    'status-list-content-bidTypeNeed',
+    'status-list-content-bidType',
+  );
+  let bidTypeClass = '';
+  let bidTypeText = '';
+  if (bidType === 'buy') {
+    bidTypeClass = bidTypeBuy;
+    bidTypeText = t('statuspage.buy');
+  } else if (bidType === 'sell') {
+    bidTypeClass = bidTypeSell;
+    bidTypeText = t('statuspage.sell');
+  } else if (bidType === 'dr') {
+    bidTypeClass = bidTypeNeed;
+    bidTypeText = t('statuspage.need');
+  }
+
+  // status box
+  const statusBoxGreen = classnames(
+    'status-list-content-statusBox-green',
+    'status-list-content-statusBox',
+  );
+  const statusBoxGrey = classnames(
+    'status-list-content-statusBox-grey',
+    'status-list-content-statusBox',
+  );
+  const statusBoxWhite = classnames(
+    'status-list-content-statusBox-white',
+    'status-list-content-statusBox',
+  );
+  const StatusBox = () => {
+    let list = [];
+    let count = 0;
+    if (status === '投標中') count = 1;
+    else if (status === '已投標') count = 2;
+    else if (status === '已得標') count = 3;
+    else if (status === '執行中') count = 4;
+    else if (status === '結算中') count = 5;
+    else if (status === '已結算') count = 6;
```

Comment:
I think it's more readable legible to have a `enum` to store this mapping.

Reply:
done